### ISSUE TITLE
Barcode marker generator

### ISSUE THREAD

**Issue by nicolocarpignoli** [AR-js-org/studio-backend#3]
We do not have source code of barcode marker generator. But don't worry: 

https://github.com/nicolocarpignoli/artoolkit-barcode-markers-collection

On that repository there are all barcode markers the user can ask for.

We just have to decide a subset of those present in there, maybe 3 sets looking for best ratio between high hamming distance/number of markers.

Then we have to include those png files on backend project and respond to a user request giving them the barcode file (base64)

**Comment by sigalor** [comment]
I have now fully reverse-engineered all the algorithms for generating barcode/matrix code markers (71e8fdaf938d544bd4e77f4efc5cf2a268fce13f); this was actually quite a fun puzzle! I also added some tests which make sure that the algorithms create exactly the same codes as the ones in the [artoolkit-barcode-markers-collection](https://github.com/nicolocarpignoli/artoolkit-barcode-markers-collection) repository.

For output, I created a simple SVG writer (5118e640174f9378712d671b22c36c37fbfc42e5) to avoid using much CPU power for writing large PNG files and to make the barcode images be arbitrarily scalable.
